### PR TITLE
Add ebut:program tests

### PR DESCRIPTION
**Pull request by matsl**
Add ebut:program tests.

### CODE REVIEW

File: test/hypb-tests.el [hunk 1]
```diff
@@ -89,5 +90,39 @@
     (should (equal (hypb:replace-match-string "\\`\\|x" "--xx--" "z")
                    "z--zz--"))))
 
+(ert-deftest hypb:program-create-but-in-buffer ()
+  "Create button with hypb:program in buffer.
+hbut:at-p does not recognise a button created within a buffer."
+  :expected-result :failed
+  (with-temp-buffer
+    (ebut:program "label" 'actypes::link-to-directory "/tmp")
+    (should (eq (hattr:get (hbut:at-p) 'actype) 'actypes::link-to-directory))
```
Comment (matsl) [inline test/hypb-tests.el:99]: hbut:at-p does not return any properties when the ebut is stored in the buffer. i.e. not in .hypb but after ^L at the end of the file. Is that expected?

File: test/hypb-tests.el [hunk 2]
```diff
@@ -89,5 +90,39 @@
...
+
+(ert-deftest hypb:program-create-link-to-file-line-and-column-but-in-file ()
+  "Create button that links to file with line and column with hypb:program in buffer."
+  (let ((test-file (make-temp-file "test-file")))
+    (unwind-protect
+        (progn
+          (find-file test-file)
+          (ebut:program "label" 'actypes::link-to-file-line-and-column test-file 2 3)
+          (should (eq (hattr:get (hbut:at-p) 'actype) 'actypes::link-to-file-line-and-column))
+          (should (equal (hattr:get (hbut:at-p) 'args) (list 'test-file 2 3)))
```
Comment (matsl) [inline test/hypb-tests.el:123]: Is this how ebut:program is supposed to work? The arg test-file is not evaluated so the variable test-file is inserted into the ebut. Does not that make ebut:program hard to use?

File: test/hypb-tests.el (hunk 1 above)
Comment (rswgnu) [reply test/hypb-tests.el:99]: Sounds like a bug that should be fixed.

File: test/hypb-tests.el (hunk 2 above)
Comment (rswgnu) [reply test/hypb-tests.el:123]: I'll have to look at that.

**Comment by matsl** [comment]
This should now make ert tests work again after ebut:program update. I had to use 'actype:: prefix though. Got undefined variable without it.

gbut was not as straight forward as I initially expected to will come back with that in another PR.

@rswgnu PR has been updated.

**Comment by rswgnu** [comment]
Please resolve the conflicts here.

**Comment by matsl** [comment]
@rswgnu Conflicts resolved.

File: test/hypb-tests.el (hunk 1 above)
Comment (rswgnu) [reply test/hypb-tests.el]: I just tested this with an explicit button created in an *unsent mail* buffer in Emacs 27.1 and it works fine.  Remember GNUS is not yet integrated with Hyperbole as a mail reader/sender, only as a news reader.  You'll have to give me a specific recipe for recreating this error to understand it.

File: test/hypb-tests.el (hunk 2 above)
Comment (rswgnu) [reply test/hypb-tests.el]: Looked at the code and tested and the args argument is evaluated now.  Maybe we had this as a macro but now this is a function so there should not be a problem.

File: test/hbut-tests.el [hunk 3]
```diff
@@ -26,21 +26,21 @@
     (unwind-protect
         (progn
           (find-file file)
-          (ebut:program "label" 'link-to-directory "/tmp")
+          (ebut:program "label" 'actypes::link-to-directory "/tmp")
```
Comment (rswgnu) [inline test/hbut-tests.el:29]: You can revert this change and other users where you add the "actypes::" prefix; it is not necessary.  I also have another improvement in my next push that will accept prettty much any format for the acton type.  After removing that prefix in ebut:program calls, you can squash and merge.

**Review by rswgnu — APPROVED** [review]

File: test/hypb-tests.el (hunk 1 above)
Comment (matsl) [reply test/hypb-tests.el]: Maybe it is an 28 problem. I will try to run it in 27.1 using docker to see if it works there.

The test case as such should work as a recipe. Create a buffer not attached to a file, run ebut-program there, inspect the result using hbut:at-p.

File: test/hypb-tests.el (hunk 2 above)
Comment (matsl) [reply test/hypb-tests.el]: The code snippet above is outdated and is left over from your comment 15 days ago. See the "outdated" mark. It was replaced after the macro change so line 123 about is not in the PR anymore like that. So yes this works as expected and the value of test-file is inserted in the button as expected.

File: test/hbut-tests.el (hunk 3 above)
Comment (matsl) [reply test/hbut-tests.el]: Yes, you are right. It works fine without that. Not sure why it stopped to work for me but now it seems fine again and I have reverted,
